[GStreamer][WebCodecs] Add VP8 temporal scalability encoding support

philn opened the pull request:
#### ab193bff49cf4e6ef3ecf8bf4ff744f86f7a41f2
<pre>
[GStreamer][WebCodecs] Add VP8 temporal scalability encoding support
<a href="https://bugs.webkit.org/show_bug.cgi?id=272381">https://bugs.webkit.org/show_bug.cgi?id=272381</a>

Reviewed by Xabier Rodriguez-Calvar.

VP8 supports the L1T1, L1T2 and L1T3 scalability modes but it is configurable only for the GStreamer
vp8enc encoder at the moment. Support for WebRTC VP8 SVC will rely on this code as well, in a follow-up patch.

* LayoutTests/platform/glib/TestExpectations:
* Source/WTF/wtf/glib/GUniquePtr.h:
* Source/WebCore/platform/graphics/gstreamer/VideoEncoderGStreamer.cpp:
(WebCore::retrieveTemporalIndex):
(WebCore::GStreamerInternalVideoEncoder::GStreamerInternalVideoEncoder):
(WebCore::GStreamerInternalVideoEncoder::initialize):
* Source/WebCore/platform/gstreamer/VideoEncoderPrivateGStreamer.cpp:
(defaultSetBitRateAllocation):
(Encoders::registerEncoder):
(videoEncoderSetBitRateAllocation):
(webkit_video_encoder_class_init):
* Source/WebCore/platform/gstreamer/VideoEncoderPrivateGStreamer.h:
(WebKitVideoEncoderBitRateAllocation::create):
(WebKitVideoEncoderBitRateAllocation::setBitRate):
(WebKitVideoEncoderBitRateAllocation::getBitRate const):
(WebKitVideoEncoderBitRateAllocation::scalabilityMode const):
(WebKitVideoEncoderBitRateAllocation::WebKitVideoEncoderBitRateAllocation):

Canonical link: <a href="https://commits.webkit.org/277367@main">https://commits.webkit.org/277367@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/711d57b2235cb2ca34aeef88771218d4594d3ab6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47109 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/26284 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/49748 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/49792 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/43158 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/49416 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/31550 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/23745 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/38375 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/47690 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/23766 "Found 1 new test failure: fast/writing-mode/english-bt-text-with-spelling-marker.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/40598 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19682 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/21153 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/41748 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/5153 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/40372 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/43491 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/42152 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/51668 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/46595 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/22133 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/18508 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/45669 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/23410 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/44675 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10461 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/24192 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/54098 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/23127 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/11098 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

calvaris reviewed (commented):
I wonder why you are using deprecated API.

philn committed:
[GStreamer][WebCodecs] Add VP8 temporal scalability encoding support
https://bugs.webkit.org/show_bug.cgi?id=272381

Reviewed by Xabier Rodriguez-Calvar.

VP8 supports the L1T1, L1T2 and L1T3 scalability modes but it is configurable only for the GStreamer
vp8enc encoder at the moment. Support for WebRTC VP8 SVC will rely on this code as well, in a follow-up patch.

* LayoutTests/platform/glib/TestExpectations:
* Source/WTF/wtf/glib/GUniquePtr.h:
* Source/WebCore/platform/graphics/gstreamer/VideoEncoderGStreamer.cpp:
(WebCore::retrieveTemporalIndex):
(WebCore::GStreamerInternalVideoEncoder::GStreamerInternalVideoEncoder):
(WebCore::GStreamerInternalVideoEncoder::initialize):
* Source/WebCore/platform/gstreamer/VideoEncoderPrivateGStreamer.cpp:
(defaultSetBitRateAllocation):
(Encoders::registerEncoder):
(videoEncoderSetBitRateAllocation):
(webkit_video_encoder_class_init):
* Source/WebCore/platform/gstreamer/VideoEncoderPrivateGStreamer.h:
(WebKitVideoEncoderBitRateAllocation::create):
(WebKitVideoEncoderBitRateAllocation::setBitRate):
(WebKitVideoEncoderBitRateAllocation::getBitRate const):
(WebKitVideoEncoderBitRateAllocation::scalabilityMode const):
(WebKitVideoEncoderBitRateAllocation::WebKitVideoEncoderBitRateAllocation):

Canonical link: https://commits.webkit.org/277367@main
